### PR TITLE
fix: DH-16595: Fix null partition filter

### LINKED ISSUE #1867: Table with a `null` partition value throws NPE
Reported with https://deephaven.atlassian.net/browse/DH-16595

Table throws NPE:
```
LogProxy.js:87 [IrisGrid] TypeError: java.lang.NullPointerException
    at NullPointerException.createError_0 [as createError] (irisapi.nocache.js:1541:10)
    at NullPointerException.initializeBackingError (irisapi.nocache.js:1329:46)
    at NullPointerException.Throwable (irisapi.nocache.js:1268:8)
    at NullPointerException.Exception (irisapi.nocache.js:1341:16)
    at NullPointerException.RuntimeException (irisapi.nocache.js:1366:16)
    at NullPointerException.JsException (irisapi.nocache.js:1515:23)
    at new NullPointerException (irisapi.nocache.js:1528:18)
    at Object.requireNonNull (irisapi.nocache.js:52632:16)
    at Object.ofString_0 [as ofString] (irisapi.nocache.js:16421:6)
    at lm.updatePartition (IrisGrid.js:1256:76)
    at IrisGrid.js:1230:17
    at Generator.next (<anonymous>)
    at gTe (IrisGrid.js:1:99)
    at o (IrisGrid.js:2:194)
isEnabled.console.error @ LogProxy.js:87
LogProxy.js:87 [IrisGridPanel] TypeError: java.lang.NullPointerException
    at NullPointerException.createError_0 [as createError] (irisapi.nocache.js:1541:10)
    at NullPointerException.initializeBackingError (irisapi.nocache.js:1329:46)
    at NullPointerException.Throwable (irisapi.nocache.js:1268:8)
    at NullPointerException.Exception (irisapi.nocache.js:1341:16)
    at NullPointerException.RuntimeException (irisapi.nocache.js:1366:16)
    at NullPointerException.JsException (irisapi.nocache.js:1515:23)
    at new NullPointerException (irisapi.nocache.js:1528:18)
    at Object.requireNonNull (irisapi.nocache.js:52632:16)
    at Object.ofString_0 [as ofString] (irisapi.nocache.js:16421:6)
    at lm.updatePartition (IrisGrid.js:1256:76)
    at IrisGrid.js:1230:17
    at Generator.next (<anonymous>)
    at gTe (IrisGrid.js:1:99)
    at o (IrisGrid.js:2:194)
```

### PR DESCRIPTION
**Pull request by vbabich**
Closes #1867

Tested in Enterprise using steps from DH-16595 - created a partitioned table and set ACL to `nullFilterGenerator()` to get the `null` value in the partition column. Creating a partitioned table with a `null` in the partition column doesn't seem possible with `db.addTablePartition`  in Enterprise.

```
t=db.t("LearnDeephaven", "StockTrades").where()
db.addPartitionedTableDefinition("<namespace>", "partTable", "Date", t.getDefinition())
test = db.t("<namespace>" , "partTable")

// Set Column ACL on `partTable` to `new NullFilterGenerator()` in swing

test = db.t("<namespace>" , "partTable")
```
Screenshot 2024-04-23 at 5 26 43 PM

Core+ Groovy console:
```
t = emptyTable(1).update("X=(String)null","Y=ii").partitionBy("X")
```
Screenshot 2024-04-23 at 5 23 55 PM

### CODE REVIEW

File: packages/iris-grid/src/IrisGridPartitionSelector.tsx [hunk 1]
```diff
@@ -214,7 +214,11 @@ class IrisGridPartitionSelector extends Component<
   getDisplayValue(index: number, value: unknown): string {
     const { model } = this.props;
 
-    if (value == null || value === '') {
+    if (value === null) {
+      return 'null';
```
Comment (mofojed) [inline packages/iris-grid/src/IrisGridPartitionSelector.tsx:218]: We display it as `(null)` in other places (like DropdownFilter, CSV export).

**Review by mofojed — CHANGES_REQUESTED** [review]
Minor display string change, can you also include how this was tested in the PR description?

**Comment by vbabich** [comment]
Changed `null` to `(null)`, added testing steps to the description.

**Comment by vbabich** [comment]
I tested this in a Core+ worker and it looks like the api returns `undefined` for a `null` partition key value. Will need to investigate a bit more.

**Comment by vbabich** [comment]
Partitioned table with a `null` key in DHE:

Screenshot 2024-04-23 at 11 34 44 AM



DHC shows an extra empty `option` element until the [core PR](https://github.com/deephaven/deephaven-core/issues/5400) is merged.

Screenshot 2024-04-23 at 2 14 57 PM

**Comment by vbabich** [comment]
Added nullish coalescing to account for `undefined` returned by the JSAPI for `null` table values

File: packages/iris-grid/src/IrisGrid.tsx [hunk 2]
```diff
@@ -2062,7 +2062,10 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
               return;
             }
             const row = data.rows[0];
-            const values = keyTable.columns.map(column => row.get(column));
+            // Core JSAPI returns undefined for null table values, IrisGridPartitionSelector expects null
```
Comment (mofojed) [inline packages/iris-grid/src/IrisGrid.tsx:2065]: Add the ticket number
```suggestion
            // Core JSAPI returns undefined for null table values, IrisGridPartitionSelector expects null (https://github.com/deephaven/deephaven-core/issues/5400)
```

**Review by mofojed — APPROVED** [review]
Minor suggestion to include the core ticket number.

**Review by mofojed — APPROVED** [review]

**Pull request merged**